src/core/server/server.ts

```diff
@@ -144,12 +147,18 @@ export class Server {
 
     const pluginsSetup = await this.plugins.setup(coreSetup);
 
+    const renderingSetup = await this.rendering.setup({
+      http: httpSetup,
+      legacyPlugins,
+      plugins: pluginsSetup,
+    });
+
     await this.legacy.setup({
-      core: { ...coreSetup, plugins: pluginsSetup },
+      core: { ...coreSetup, plugins: pluginsSetup, rendering: renderingSetup },
       plugins: mapToObject(pluginsSetup.contracts),
     });
 
-    this.registerCoreContext(coreSetup);
+    this.registerCoreContext(coreSetup, renderingSetup);
```

Comment:
(outside of the scope of the PR) we have a chicken/egg condition between `InternalCoreSetup` creation and the services that actually relies on it here (plugins and now rendering)....

Reply:
Agreed, this was weird to work around.